DEVELOPMENT.md

```diff
@@ -0,0 +1,20 @@
+Developer docs
+==================================
+
+## 📝 Prerequisites
+* ffmpeg
+* Python (Current required version van be found in [pyproject.toml](https://github.com/music-assistant/server/blob/main/pyproject.toml#LL11C14-L11C14))
+* [Python venv](https://docs.python.org/3/library/venv.html)
+
+It is recommended to use Visual Studio Code as your IDE, since launch files to start Music Assistant are provided as part of the repository. Furthermore, the current code base is not verified to work on a native Windows machine. If you would like to develop on a Windows machine, install [WSL2](https://code.visualstudio.com/blogs/2019/09/03/wsl2) to increase your swag-level 🤘.
+
+## 🚀 Setting up your development environment
+With this repostiory cloned locally, execute the following commands in a terminal from the root of your repository:
+* `python -m venv .venv` (create a new separate virtual environment to nicely separate the project dependencies)
+* `source .venv/bin/activate` (activate the virtual environment)
+* `pip install -r requirements_all.txt` (install the project's dependencies)
+* Hit (Fn +) F5 to start Music Assistant locally 🎉
+
+
+## 🎵 Building your own Music Provider
```

Comment:
Yeah, and maybe also mention we have Player Providers, Metadata and Plugin providers